presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -918,6 +918,27 @@ public void testInsertIntoPartitionedTable()
         }
     }
 
+    @Test
+    public void testInsertIntoSpecialPartitionName()
+    {
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+        // Clean up if temporary table already exists.
+        dropTableIfExists(tmpTableName);
+
+        try {
+            getQueryRunner().execute(getSession(), String.format("CREATE TABLE %s (name VARCHAR, nationkey VARCHAR) WITH (partitioned_by = ARRAY['nationkey'])", tmpTableName));
+            Session writeSession = buildSessionForTableWrite();
+
+            // Test special character in partition name, without proper handling, it would throw 'Invalid partition spec: nationkey=A/B'
+            getQueryRunner().execute(writeSession, String.format("INSERT INTO %s VALUES (\'name\', \'A/B\')", tmpTableName));
```

Comment:
It might be good to have a Strings[] to list all the special characters to test here. Thanks!